### PR TITLE
English Dictionary HW

### PR DESCRIPTION
**Pull request by SheyReiff**
Screenshot 2024-04-03 at 11 00 14 AM

### CODE REVIEW

File: src/main/java/reiff/dictionary/DictionaryFrame.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                    "Error", JOptionPane.ERROR_MESSAGE);
+        } catch (CsvValidationException e) {
+            throw new RuntimeException(e);
+        }
+
+        initializeGui();
+    }
+
+    private void initializeGui() {
+        final JFrame frame = new JFrame("DictionaryFrame");
```
Comment (andrewoid) [inline src/main/java/reiff/dictionary/DictionaryFrame.java]: Is this how I taught you to create jframes?

File: src/main/java/reiff/dictionary/DictionaryFrame.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    public DictionaryFrame() {
+        try {
+            dictionary = new EnglishDictionary();
+        } catch (IOException e) {
+            e.printStackTrace();
+            JOptionPane.showMessageDialog(null, "Error loading the dictionary: " + e.getMessage(),
+                    "Error", JOptionPane.ERROR_MESSAGE);
+        } catch (CsvValidationException e) {
+            throw new RuntimeException(e);
+        }
```
Comment (andrewoid) [inline src/main/java/reiff/dictionary/DictionaryFrame.java]: You can do the same thing for both

File: src/main/java/reiff/dictionary/DictionaryFrame.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            public void keyTyped(KeyEvent e) {
+            }
+
+            @Override
+            public void keyPressed(KeyEvent e) {
+            }
+
+            @Override
+            public void keyReleased(KeyEvent e) {
+                String word = wordField.getText().trim().toLowerCase();
```
Comment (andrewoid) [inline src/main/java/reiff/dictionary/DictionaryFrame.java]: ToLowerCase is already done inside of getDefinition

File: src/test/java/reiff/dictionary/EnglishDictionaryTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @Test
+    public void wordDoesNotExist() throws CsvValidationException, IOException {
+        // given
+        EnglishDictionary dictionary = new EnglishDictionary();
+
+        // when
+        List<String> definitions = dictionary.getDefinition("zip");
+
+        // then
+        assertArrayEquals(new String[]{}, definitions.toArray(new String[0]));
```
Comment (andrewoid) [inline src/test/java/reiff/dictionary/EnglishDictionaryTest.java]: AssertEquals works on Lists

File: src/main/java/reiff/dictionary/DictionaryFrame.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            dictionary = new EnglishDictionary();
+        } catch (IOException | CsvValidationException e) {
+            e.printStackTrace();
+            JOptionPane.showMessageDialog(null, "Error loading the dictionary: " + e.getMessage(),
+                    "Error", JOptionPane.ERROR_MESSAGE);
+        }
+
+        final JFrame frame = new JFrame("DictionaryFrame");
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setSize(400, 300);
```
Comment (andrewoid) [inline src/main/java/reiff/dictionary/DictionaryFrame.java]: Again. This isn't how I taught you to do this

Comment (SheyReiff) [reply src/main/java/reiff/dictionary/DictionaryFrame.java]: Are you referring to the order of the lines?

Comment (andrewoid) [reply src/main/java/reiff/dictionary/DictionaryFrame.java]: I'm referring to nowhere do you don't me calling "new JFrame"

Comment (SheyReiff) [reply src/main/java/reiff/dictionary/DictionaryFrame.java]: I understand now, thank you!

File: src/main/java/reiff/dictionary/DictionaryFrame.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        wordField.setPreferredSize(new Dimension(getWidth(), 30));
+        definitionsArea = new JTextArea();
+        definitionsArea.setEditable(false);
+
+        north.add(wordField);
+
+        main.add(new JScrollPane(definitionsArea), BorderLayout.CENTER);
+
+        setupListeners();
+        setVisible(true);
```
Comment (andrewoid) [inline src/main/java/reiff/dictionary/DictionaryFrame.java]: You can remove this because setVisible() is done in the main()

**Review by andrewoid — APPROVED** [review]

**Pull request merged**